tests/manage/z_cluster/nodes/test_check_pod_status_after_two_nodes_shutdown_recovery.py

```diff
@@ -0,0 +1,161 @@
+import logging
+import pytest
+import time
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.framework.testlib import (
+    ManageTest, tier4c, ignore_leftovers
+)
+from tests.sanity_helpers import Sanity
+from ocs_ci.ocs.node import (
+    wait_for_nodes_status, get_node_objs, get_typed_nodes
+)
+from ocs_ci.utility.retry import retry
+from ocs_ci.ocs.exceptions import CommandFailed, ResourceWrongStatusException
+from ocs_ci.ocs.resources.pod import get_all_pods
+from tests.helpers import get_node_running_pod
+
+log = logging.getLogger(__name__)
+
+
+@ignore_leftovers
+@tier4c
+class TestOCSWorkerNodeShutdown(ManageTest):
+    """
+    Test case validate both the MDS pods not running on same node
+    post shutdown and recovery
+
+    """
+
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+
+        """
+
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request, nodes):
+        """
+        Restart nodes that are in status NotReady
+        for situations in which the test failed in between
+
+        """
+
+        def finalizer():
+
+            # Validate all nodes are in READY state
+            not_ready_nodes = [
+                n for n in get_node_objs() if n
+                .ocp.get_resource_status(n.name) == constants.NODE_NOT_READY
+            ]
+            log.warning(
+                f"Nodes in NotReady status found: {[n.name for n in not_ready_nodes]}"
+            )
+            if not_ready_nodes:
+                nodes.restart_nodes_by_stop_and_start(not_ready_nodes)
+                wait_for_nodes_status()
+
+            log.info("All nodes are in Ready status")
+
+        request.addfinalizer(finalizer)
+
+    def list_of_nodes_running_pods(self, selector):
+        """
+        The function returns the list of nodes for the given selector
+
+        Args:
+            selector (str): The resource selector to search with
+
+        Returns:
+            list: Pods_running_nodes
+
+        """
+        pod_obj_list = get_all_pods(
+            namespace=defaults.ROOK_CLUSTER_NAMESPACE, selector=[selector]
+        )
+        pods_running_nodes = []
+        for pod in pod_obj_list:
+            pods_running_nodes.append(
+                get_node_running_pod(
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE, pod_name=pod.name
+                )
+            )
+
+        log.info(f"{selector} running on nodes {pods_running_nodes}")
+        return pods_running_nodes
+
+    @pytest.mark.polarion_id("OCS-1278")
+    @pytest.mark.polarion_id("OCS-2315")
```

Comment:
The scenario is same, it is just that we have included one more additional check to the existing test.
Can we combine it to a single test?

Reply:
Done